aead/src/lib.rs

```diff
@@ -68,96 +75,220 @@ impl fmt::Display for Error {
 
 impl core::error::Error for Error {}
 
-/// Nonce: single-use value for ensuring ciphertexts are unique
-pub type Nonce<A> = Array<u8, <A as AeadCore>::NonceSize>;
-
-/// Tag: authentication code which ensures ciphertexts are authentic
-pub type Tag<A> = Array<u8, <A as AeadCore>::TagSize>;
-
-/// Authenticated Encryption with Associated Data (AEAD) algorithm core trait.
-///
-/// Defines nonce, tag, and overhead sizes that are consumed by various other
-/// `Aead*` traits.
-pub trait AeadCore {
+/// Authenticated Encryption with Associated Data (AEAD) algorithm trait.
+pub trait Aead {
     /// The length of a nonce.
     type NonceSize: ArraySize;
```

Comment:
Maybe we should use `IvSizeUser` here instead (after renaming it to `NonceSizeUser`)? We will get the RNG methods and `Nonce` alias "for free". It also would work a bit nicer with the stream traits.